os/kernel/semaphore/sem_list.c

```diff
@@ -43,11 +47,30 @@ sq_queue_t g_sem_list;
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+static void sem_addlist(FAR sem_t *sem, sq_queue_t *list)
+{
+	sem_t *sem_ptr;
+	irqstate_t flags;
+
+	flags = irqsave();
+
+	sem_ptr = (sem_t *)sq_peek(list);
+	while (sem_ptr) {
```

Comment:
I'm worried about this code. Because sometimes this list can be very long, then it takes so much time.
How about using another way to check that sem is registered or not, like hash?